src/printer.js

```diff
@@ -3297,6 +3303,19 @@ function printArrayItems(path, options, printPath, print) {
   return concat(printedElements);
 }
 
+function isParserKeepingEntities(node, options) {
```

Comment:
Here comes the new helper function, depending on the node type it performs a simple check with adjustments. If parser wouldn't convert it to an entity, we should not do it too.